gsuite_reports_rules/gsuite_drive_visibility_change.py

```diff
@@ -1,42 +1,123 @@
 from panther_base_helpers import deep_get
 from panther_base_helpers import gsuite_parameter_lookup as param_lookup
 
+EXCLUDED_DOMAINS = ["runpanther.io"]
+
+VISIBILITY = ["people_with_link", "public_on_the_web", "shared_externally", "unknown"]
+
+DOC_TITLE = "<UNKNOWN_TITLE>"
+TARGET_USER_EMAIL = "<UNKNOWN_USER>"
+TARGET_DOMAIN = "<UNKNOWN_DOMAIN>"
+ACCESS_SCOPE = "<UNKNOWN_ACCESS_SCOPE>"
+NEW_VISIBILITY = "<UNKNOWN_VISIBILITY>"
+
 
 def rule(event):
     if deep_get(event, "id", "applicationName") != "drive":
         return False
 
+    #########
+    # for target_domain
+    change_document_visibility = False
+
+    for details in event.get("events", [{}]):
+        if (
+            details.get("type") == "acl_change"
+            and details.get("name") == "change_document_visibility"
+            and param_lookup(details.get("parameters", {}), "new_value") != ["private"]
+            and param_lookup(details.get("parameters", {}), "target_domain") not in EXCLUDED_DOMAINS
+            and param_lookup(details.get("parameters", {}), "visibility") in VISIBILITY
+        ):
+            global TARGET_DOMAIN
+            TARGET_DOMAIN = param_lookup(details.get("parameters", {}), "target_domain")
+            global NEW_VISIBILITY
+            NEW_VISIBILITY = param_lookup(details.get("parameters", {}), "visibility")
+            global DOC_TITLE
+            DOC_TITLE = param_lookup(details.get("parameters", {}), "doc_title")
```

Comment:
I find this slightly more readable:

```suggestion
            global TARGET_DOMAIN, NEW_VISIBILITY, DOC_TITLE
            TARGET_DOMAIN = param_lookup(details.get("parameters", {}), "target_domain")
            NEW_VISIBILITY = param_lookup(details.get("parameters", {}), "visibility")
            DOC_TITLE = param_lookup(details.get("parameters", {}), "doc_title")
```